java/code/src/com/suse/manager/virtualization/VirtualMachineRestarterSalt.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.virtualization;
+
+import com.suse.manager.webui.services.iface.VirtManager;
+import com.suse.manager.webui.services.iface.VirtualMachineRestarter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Salt implementation for the VM restart handler
+ */
+public class VirtualMachineRestarterSalt implements VirtualMachineRestarter {
+
+    private static VirtualMachineRestarterSalt instance;
+
+    private final VirtManager virtManager;
+    private final Object lock = new Object();
```

Comment:
you can just lock use the ArrayList object as lock and don't need a separate object.

Reply:
Indeed, will change that